python/grizzly/grizzly/grizzly.py

```diff
@@ -5,6 +5,367 @@
 from lazy_op import LazyOpResult, to_weld_type
 from weld.weldobject import *
 
+class DataFrameWeldLoc:
+    def __init__(self, df):
+        self.df = df
+
+    def __getitem__(self, key):
+        if isinstance(key, SeriesWeld):
+            # We're going to assume that the first column in these dataframes
+            # is an index column. This assumption does not hold throughout grizzly,
+            # so we should fix that moving forward.
+            index_expr = grizzly_impl.get_field(self.df.expr, 0)
+            if self.df.is_pivot:
+                index_type, pivot_type, column_type = self.df.column_types
+                index_elem_type = index_type.elemType
+                index_expr_predicate = grizzly_impl.isin(index_expr, key.expr, index_elem_type)
+                return DataFrameWeldExpr(
+                    grizzly_impl.pivot_filter(
+                        self.df.expr,
+                        index_expr_predicate
+                    ),
+                    self.df.column_names,
+                    self.df.weld_type,
+                    is_pivot=True
+                )
+            # TODO : Need to implement for non-pivot tables
+        raise Exception("Cannot invoke getitem on an object that is not SeriesWeld")
+
+class WeldLocIndexer:
+    def __init__(self, grizzly_obj):
+        # If index_type field of grizzly_obj is None
+        # then we assume normal 0 - 1 indexing
+        self.grizzly_obj = grizzly_obj
+
+    def __getitem__(self, key):
+        if isinstance(self.grizzly_obj, SeriesWeld):
+            series = self.grizzly_obj
+            if isinstance(key, SeriesWeld):
+                if series.index_type is not None:
+                    index_expr = grizzly_impl.get_field(series.expr, 0)
+                    column_expr = grizzly_impl.get_field(series.expr, 1)
+                    zip_expr = grizzly_impl.zip_columns([index_expr, column_expr])
+                    predicate_expr = grizzly_impl.isin(index_expr, key.expr, series.index_type)
+                    filtered_expr = grizzly_impl.filter(
+                        zip_expr,
+                        predicate_expr
+                    )
+                    unzip_expr = grizzly_impl.unzip_columns(
+                        filtered_expr,
+                        [series.index_type, series.weld_type]
+                    )
+                    return SeriesWeld(
+                        unzip_expr,
+                        series.weld_type,
+                        series.df,
+                        series.column_name,
+                        series.index_type,
+                        series.index_name
+                    )
+
+            # TODO : Need to implement for non-pivot tables
+        raise Exception("Cannot invoke getitem on non SeriesWeld object")
+
+class DataFrameWeldExpr:
```

Comment:
Why do we have so many of these classes

Reply:
I found it difficult dealing with DataFrameWeld which kept track of un-materialized columns and materialized columns. Instead treating dataframes as a zip of columns from the onset makes it very easy to deal with. We should really just merge these two classes.